packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m

```diff
@@ -87,13 +88,25 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     [[GIDSignIn sharedInstance] signOut];
     result(nil);
   } else if ([call.method isEqualToString:@"disconnect"]) {
-    [_accountRequests insertObject:result atIndex:0];
-    [[GIDSignIn sharedInstance] disconnect];
+    if ([self setAccountRequest:result]) {
+      [[GIDSignIn sharedInstance] disconnect];
+    }
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
+- (BOOL)setAccountRequest:(FlutterResult)request {
+  if (_accountRequest != nil) {
+    request([FlutterError errorWithCode:@"concurrent-requests"
```

Comment:
This is different than the error you'll get in Java.  Makes me wonder if we should change `checkAndSetPendingOperation()` to throw a private static Exception class that's caught in `onMethodCall()` and translated to a call to `result.error()`